ZipEntry.cs

```diff
@@ -386,14 +387,16 @@ void DoExtract (IntPtr zipFile, string destinationPath, FileMode outputFileMode,
 
 		void DoExtract (IntPtr zipFile, Stream destinationStream, EntryExtractEventArgs args)
 		{
-			var buf = new byte [ReadBufSize];
-
-			long nread;
-
-			while ((nread = Native.zip_fread (zipFile, buf, (ulong)buf.Length)) > 0) {
-				destinationStream.Write (buf, 0, (int)nread);
-				args.ProcessedSoFar += (ulong)nread;
-				OnExtract (args);
+			var buf = ArrayPool<byte>.Shared.Rent (ReadBufSize);
+			try {
+				long nread;
+				while ((nread = Native.zip_fread (zipFile, buf, (ulong)buf.Length)) > 0) {
```

Comment:
Here (and some other places), if `ArrayPool` gives us a bigger `byte[]` the existing code seemed OK because it was using `buf.Length`. We can take advantage of the bigger buffer if we happened to get one.